src/jemalloc.c

```diff
@@ -2068,13 +2069,22 @@ imalloc_body(static_opts_t *sopts, dynamic_opts_t *dopts, tsd_t *tsd) {
 			    <= SC_LARGE_MAXCLASS);
 		}
 	} else {
+		if (sopts->bump_empty_aligned_alloc) {
+			if (unlikely(size == 0)) {
+				size = 1;
+			}
+		}
 		usize = sz_sa2u(size, dopts->alignment);
 		dopts->usize = usize;
 		if (unlikely(usize == 0
 		    || usize > SC_LARGE_MAXCLASS)) {
 			goto label_oom;
 		}
 	}
+	/* Validate the user input. */
```

Comment:
I don't think this needs to move? assert_nonempty_alloc should just be false for posix_memalign and aligned_alloc.

Reply:
Moved this as in the comment for assert_nonempty_alloc we say “Whether to assert that allocations are not of size 0 (after any bumping).” So wanted to make sure all the bumping are done before the assertion. Does that make sense or a bit too much?

Reply:
Oh I see. Yeah that makes sense.